test/extensions/filters/http/ext_authz/ext_authz_grpc_fuzz_test.cc

```diff
@@ -0,0 +1,115 @@
+
+#include "source/common/common/assert.h"
+#include "source/extensions/filters/common/ext_authz/ext_authz_http_impl.h"
+
+#include "test/extensions/filters/common/ext_authz/test_common.h"
+#include "test/extensions/filters/http/common/fuzz/http_filter_fuzzer.h"
+#include "test/extensions/filters/http/ext_authz/ext_authz_fuzz.pb.h"
+#include "test/extensions/filters/http/ext_authz/ext_authz_fuzz_lib.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/grpc/mocks.h"
+
+#include "gmock/gmock.h"
+
+using Envoy::Extensions::Filters::Common::ExtAuthz::TestCommon;
+using envoy::extensions::filters::http::ext_authz::ExtAuthzTestCase;
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExtAuthz {
+namespace {
+
+std::unique_ptr<envoy::service::auth::v3::CheckResponse>
+makeGrpcCheckResponse(const Grpc::Status::WellKnownGrpcStatus status) {
+  auto response = std::make_unique<envoy::service::auth::v3::CheckResponse>();
+  response->mutable_status()->set_code(status);
+  // TODO: We only add the response status.
+  // Add fuzzed inputs for headers_to_(set/append/add), body, status_code to the Response.
+  return response;
+}
+
+Grpc::Status::WellKnownGrpcStatus
+resultCaseToGrpcStatus(const ExtAuthzTestCase::AuthResult result) {
+  switch (result) {
+    PANIC_ON_PROTO_ENUM_SENTINEL_VALUES;
+  case ExtAuthzTestCase::OK:
+    return Grpc::Status::WellKnownGrpcStatus::Ok;
+  case ExtAuthzTestCase::ERROR:
+    return Grpc::Status::WellKnownGrpcStatus::Internal;
+  case ExtAuthzTestCase::DENIED:
+    return Grpc::Status::WellKnownGrpcStatus::PermissionDenied;
+  }
+  PANIC_DUE_TO_CORRUPT_ENUM;
+}
+
+class ReusableGrpcClientFactory {
+public:
+  ReusableGrpcClientFactory()
+      : internal_grpc_mock_client_(std::make_shared<NiceMock<Grpc::MockAsyncClient>>()) {
+    ON_CALL(*internal_grpc_mock_client_, sendRaw(_, _, _, _, _, _))
+        .WillByDefault(
+            Invoke([&](absl::string_view, absl::string_view, Buffer::InstancePtr&& serialized_req,
+                       Grpc::RawAsyncRequestCallbacks&, Tracing::Span&,
+                       const Http::AsyncClient::RequestOptions&) -> Grpc::AsyncRequest* {
+              envoy::service::auth::v3::CheckRequest check_request;
+              EXPECT_TRUE(check_request.ParseFromString(serialized_req->toString()))
+                  << "Could not parse serialized check request";
+
+              // TODO: Query the request header map in HttpFilterFuzzer to test
+              // headers_to_(add/remove/append).
+              // TODO: Test check request attributes against config
+              // and filter metadata.
+              ENVOY_LOG_MISC(trace, "Check Request attributes {}",
+                             check_request.attributes().DebugString());
+
+              if (status_ == Grpc::Status::WellKnownGrpcStatus::Ok) {
+                grpc_client_->onSuccess(makeGrpcCheckResponse(status_), mock_span_);
```

Comment:
Good job to mock the response from client layer, instead of directly call filter layer code.  This will expand the fuzzer code coverage to ext_authz gRPC client code.  And same for ext_authz HTTP client code.